scripts/shellcheck.sh

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env bash
+
+CYAN="\e[36m"
+YELLOW="\e[33m"
+BLUE="\e[34m"
+RED="\e[31m"
+GREEN="\e[32m"
+RST_COLOR="\e[0m"
+
+ROOT="$(cd "$(dirname "$0")"/.. && pwd)"
+SHELLCHECK_REPORT="${ROOT}/shellcheck-report.txt"
+START_TIME=$(date "+%Y-%m-%d %H:%M:%S")
+
+show_usage() {
+    cat << EOF
+Usage: $0 [options]
+
+Options:
+    -h, --help               print this help and exit
+    -e, --exclude-code       exclude specific error code, can be repeated
+    -S, --severity=LEVEL     set severity level (info, warning, error)
+    -v, --view               view shellcheck report
+
+Usage: ./scripts/shellcheck.sh [-e|--exclude-code=CODE] [-S|--severity=LEVEL] [-v|--view]
+
+example: ./scripts/shellcheck.sh -e SC1091 -e SC1090 -S warning -v
+
+EOF
+}
+
+parse_cli_args() {
+    echo; echo "Command line arguments: $0 $*"; echo
+
+    SEVERITY=
+    EXCLUDE_CODES=
+    VIEW=
+
+    _help=
+    _invalid_arg_count=0
+    while [[ $# -gt 0 ]]; do
+        optarg=
+        _err=0
+        shft_cnt=1
+        if [[ "$1" = '--' ]]; then
+            break
+        elif [[ "$1" =~ ^-- && $# -gt 1 && ! "$2" =~ ^- ]]; then
+            optarg="$2"
+            shft_cnt=2
+        elif [[ "$1" =~ ^-[^-] && $# -gt 1 && ! "$2" =~ ^- ]]; then
+            optarg="$2"
+            shft_cnt=2
+        elif [[ "$1" =~ ^-[^-] ]]; then
+            optarg="${1/??/}"
+        fi
+
+        case "$1" in
+            -S*|--severity*)
+                # don't let the user enter -S LEVEL more than once
```

Comment:
This is bad practice; it's much better to give higher-precedence to later-specified arguments (i.e., `--level=warn --level=error --level=info` should set `level` equal to `info`). This makes it so that aliases and functions can hard-code arguments to programs but still allow the caller to override them (creating an ad-hoc default-mechanism). Plus, it's actually just simpler: each specification of the argument just overwrites the previous value, and that's it!

Reply:
ACK, I'll update the logic to allow this